### PR TITLE
Begins DonorsChooseBot refactor

### PR DESCRIPTION
**Pull request by aaronschachter**
#### What's this PR do?
Begins a `/v1/donorschoosebot` endpoint to eventually deprecate use of `/v1/chatbot?bot_type=donorschoosebot`. This endpoint will post a donation if all the expected Mobile Commons Profile fields are passed in the `req.body`, but doesn't save anything to the user's  Mobile Commons Profile. Opening this now to help keep the pull request reviewable.

* Adds new `app/routes/donorschoosebot` to handle logic of inspecting Mobile Commons Profile, sent message from User, and determining how to respond.
    * Uses promises to avoid callback hell (similar to CampaignBot makeover in #632)

* Adds `lib/donorschoose.js` for lint -- adds helper `superagent` functions for DonorsChoose API requests

* Adds `DONORSCHOOSE_API_BASEURI` and `DONORSCHOOSE_API_SECURE_BASEURI` to avoid hardcoding URL's for use with new route

#### How should this be reviewed?
Post to `/v1/donorschoosebot` locally with expected params defined in [chatbot docs](https://github.com/DoSomething/gambit/blob/develop/documentation/endpoints/chatbot.md), if verify ask/invalid messages are sent, and a donation is created if all fields are passed.

#### Any background context you want to provide?

The only time we post to DonorsChooseBot is when a user has completed a SMS Game, so it makes sense to move it out of our `chatbot` endpoint entirely and into its own `donorschoosebot` endpoint.

Next steps:
* [ ] Save data to Mobile Commons Profile and deliver messages
* [ ] Update documentation
* [ ] Once live, remove the `DonorsChooseBotController`. Our `package.json` can simply lint our entire `app` directory instead of specifying each subdirectory

#### Relevant tickets
#584 

#### Checklist
- [x] Tested on staging.

### CODE REVIEW

File: app/routes/donorschoosebot.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  let incomingMessage = req.body.args;
+  if (!incomingMessage) {
+    return helpers.sendResponse(res, 422, 'Missing required args.');
+  }
+
+  incomingMessage = helpers.getFirstWord(incomingMessage);
+  logger.debug(`dcbot incomingMessage:${incomingMessage}`);
+
+  const profileFieldName = `profile_${COUNT_FIELD}`;
+  const numDonations = req.body[profileFieldName];
```
Comment (sergii-tkachenko) [inline app/routes/donorschoosebot.js]: If `req.body[profileFieldName] == undefined`?

File: app/routes/donorschoosebot.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  if (numDonations >= MAX_DONATIONS_ALLOWED) {
+    return sendResponse(res, 200, 'max_donations_reached');
+  }
+
+  // If start param passed, initiate conversation, ignoring incomingMessage.
+  const prompt = req.query.start;
+
+  if (!req.body.profile_postal_code) {
+    if (prompt) {
+      return sendResponse(res, 200, 'ask_zip');
```
Comment (sergii-tkachenko) [inline app/routes/donorschoosebot.js]: Why 200? It's an error, right? Maybe 422 or 400?

**Review by sergiitk — CHANGES_REQUESTED** [review]

File: app/routes/donorschoosebot.js (hunk 2 above)
Comment (aaronschachter) [inline app/routes/donorschoosebot.js]: It's not meant to be an error -- Mobile Commons posts an entire Profile in `req.body` -- this  check is to see if there's a value stored in that field. If not, we send a message to User asking for zip code. Send a 200 code back to Mobile Commons mData saying Yup I know how to respond to this, I sent a `profile_update` request.

Comment (sergii-tkachenko) [inline app/routes/donorschoosebot.js]: Maybe then just explain it in the comment.

File: app/routes/donorschoosebot.js (hunk 1 above)
Comment (sergii-tkachenko) [inline app/routes/donorschoosebot.js]: I thought more like
```js
const numDonations = req.body[profileFieldName] ? Number(req.body[profileFieldName]) : -1;
```

**Review by sergiitk — APPROVED** [review]
👍

**Pull request merged**